packages/core/src/external/mpi/normalize-patient.ts

```diff
@@ -0,0 +1,167 @@
+        newAddress.addressLine2 = normalizeAddress(addr.addressLine2);
+      }
+      return newAddress;
+    }),
+  };
+  return handleDefaultValues(normalizedPatient);
+}
+
+/**
+ * The normalizeString function takes a string as input, removes leading and trailing whitespace,
+ * converts all characters to lowercase, and removes any apostrophes or hyphens.
+ * @param {string} str - The `str` parameter is a string that represents the input string that needs to
+ * be normalized.
+ * @returns a normalized version of the input string.
+ */
+function normalizeString(str: string): string {
+  return str.trim().toLowerCase().replace(/['-]/g, "");
+}
+
+/**
+ * Normalizes an email address by removing leading and trailing spaces, converting all characters to lowercase,
+ * and handling common domain typos.
+ *
+ * @param email - The email address to be normalized.
+ * @returns The normalized email address.
+ */
+function normalizeEmail(email: string): string {
+  return email.trim().toLowerCase();
+}
+
+/**
+ * Normalizes a phone number by removing all non-numeric characters and, if applicable, removing the country code.
+ * @param phoneNumber - The phone number to be normalized.
+ * @returns The normalized phone number as a string.
+ */
+function normalizePhoneNumber(phoneNumber: string): string {
+  const normalizedNumber = phoneNumber.replace(/\D/g, "");
+
+  if (normalizedNumber.startsWith("1") && normalizedNumber.length === 11) {
+    return normalizedNumber.substring(1);
+  }
+
+  return normalizedNumber;
+}
+
+/**
+ * The function `normalizeAddress` takes a string representing an address and replaces common street
+ * suffixes with their abbreviated forms.
+ * @param {string} address - The `address` parameter is a string that represents a street address.
+ * @returns The function `normalizeAddress` returns a string.
+ */
+function normalizeAddress(address: string): string {
+  const suffixes: Record<string, string> = {
+    street: "st",
+    avenue: "ave",
+    boulevard: "blvd",
+    drive: "dr",
+    road: "rd",
+    terrace: "ter",
+    place: "pl",
+    lane: "ln",
+    court: "ct",
+    circle: "cir",
+    highway: "hwy",
+    parkway: "pkwy",
+  };
+
+  address = address.trim().toLowerCase().replace(/['-.]/g, "");
+  const words: string[] = address.split(" ");
+
+  for (let i = 0; i < words.length; i++) {
+    const word = words[i];
+    if (word) {
+      const suffix = suffixes[word];
+      if (suffix) {
+        words[i] = suffix;
+      }
+    }
+  }
+  return words.join(" ");
+}
+
+export function splitName(name: string): string[] {
+  // splits by comma delimiter and filters out empty strings
+  return name.split(/[\s,]+/).filter(str => str);
+}
+// Define default values for each field
+const defaultValues = {
+  firstName: "john",
+  lastName: "doe",
+  address: [{ addressLine1: "123 main street", city: "anytown", zip: "00000" }],
+  contact: [{ email: "example@example.com", phone: "0000000000" }],
+};
```

Comment:
this isn't an exhaustive list, but I have personally seen all of these. Amenable to to holding off doing anything with default values until we learn more about our MPI performance in the wild.